client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py

```diff
@@ -57,12 +57,29 @@ def get_destination_path(
     dest_path: Optional[str],
     recursive_from_prefix: Optional[str] = None,
 ) -> str:
+    """
+    Get the destination path for a file download.
+
+    Args:
+        url (str): The URL to download
+        dest_path (str): The destination path the files will download to
+        recursive_from_prefix (str): The prefix to use for recursive downloads
+
+    Returns:
+        str: The destination path for the file download
+
+    Raises:
+        ValueError: If the destination path does not exist and recursive downloads are enabled
+
+    """
     if not dest_path:
         dest_path = os.getcwd()
     dest_path = os.path.abspath(dest_path)
 
     if not os.path.isdir(dest_path) and recursive_from_prefix:
-        raise ValueError("Recursive downloads require a base directory")
+        raise ValueError(
+            f"The destination path '{dest_path}' does not exist. Recursive downloads require an existing base directory.",
```

Comment:
@kandarpksk what are your thoughts on the modified error message.

Reply:
I just read recent comments in the [issue](https://github.com/chanzuckerberg/cryoet-data-portal/issues/887), and think we should just create the missing folder in all cases.

If we don't go with that, I find the updated message still confusing. I think the language we use should be from the perspective of the API user, who is either downloading a file (with say, download_mrcfile) or a folder (with say, download_everything). In which case is the download recursive? (I'd ordinarily assume that the folder download is recursive, but from [Manasa's comment](https://github.com/chanzuckerberg/cryoet-data-portal/issues/887#issuecomment-2251089496), it seems to me that in that case, this error is not thrown!)